src/lib/shared/networkPersistenceCore.mjs

```
@@ -151,20 +184,62 @@ export async function writeNetworkFile(data) {
-		await access(directory, constants.W_OK);
+		await access(source, constants.F_OK);
 	} catch {
-		return false;
+		return {
+			writable: true,
+			reason: null
+		};
+	}
+
+	try {
+		await access(source, constants.R_OK | constants.W_OK);
```

ESLint

[error] Unexpected use of '|'.

(no-bitwise)